arangod/Cluster/ClusterInfo.cpp

```diff
@@ -3287,17 +3294,14 @@ Result ClusterInfo::createCollectionsCoordinator(
       << " isNewDatabase: " << isNewDatabase
       << " first collection name: " << infos[0].name;
 
-  // The following three are used for synchronization between the callback
+  // The following are used for synchronization between the callback
   // closure and the main thread executing this function. Note that it can
   // happen that the callback is called only after we return from this
   // function!
-  auto dbServerResult =
-      std::make_shared<std::atomic<std::optional<ErrorCode>>>(std::nullopt);
-  auto nrDone = std::make_shared<std::atomic<size_t>>(0);
-  auto errMsg = std::make_shared<std::string>();
-  auto cacheMutex = std::make_shared<std::mutex>();
-  auto cacheMutexOwner = std::make_shared<std::atomic<std::thread::id>>();
-  auto isCleaned = std::make_shared<bool>(false);
+  // Also note: We will use this guarded structure to protect access into state
+  // entries inside the infos list, as tehy can be modified concurrently.
+  auto report = std::make_shared<Guarded<CreateCollectionReport>>(
+      CreateCollectionReport{});
```

Comment:
It will call default ctor
```suggestion
  auto report = std::make_shared<Guarded<CreateCollectionReport>>();
```